### PR TITLE
Smallpatches

### PR DESCRIPTION
**Pull request by InfusOnWoW**

### CODE REVIEW

File: WeakAuras/BuffTrigger.lua [hunk 1]
```diff
@@ -1551,6 +1558,7 @@ function BuffTrigger.Modernize(data)
         end
       end
     end
+
```
Comment (emptyrivers) [inline WeakAuras/BuffTrigger.lua]: Excess newline

File: WeakAuras/Prototypes.lua [hunk 2]
```diff
@@ -4036,6 +4065,14 @@ WeakAuras.event_prototypes = {
...
         conditionType = "string",
       },
+      {
+        name = "spellId",
+        display = L["Spell Id"],
+        type = "spell" ,
+        enable = function(trigger) return not(trigger.use_inverse) end,
+        test = "GetSpellInfo([[%s]]) == spell"
+      },
+
```
Comment (emptyrivers) [inline WeakAuras/Prototypes.lua]: Extra newline

File: WeakAurasOptions/OptionsFrames/IconPicker.lua [hunk 3]
```diff
@@ -33,7 +33,13 @@ local function ConstructIconPicker(frame)
     local distances = {};
     local names = {};
 
-    subname = tonumber(subname) and GetSpellInfo(tonumber(subname)) or subname;
+    if (tonumber(subname)) then
+      local spellId = tonumber(subname);
+      -- Work around special numbers such as inf and nan
+      if (abs(spellId) < math.huge and spellId == spellId) then
```
Comment (emptyrivers) [inline WeakAurasOptions/OptionsFrames/IconPicker.lua]: As I recall, blizzard has a very nonstandard approach to NaN, and guarding against it via spellId == spellId won’t work. Has this changed?

File: WeakAuras/WeakAuras.lua [hunk 4]
```diff
@@ -1470,8 +1475,9 @@ function WeakAuras.ScanForLoads(self, event, arg1)
   for id, data in pairs(db.displays) do
     if (data and not data.controlledChildren) then
       local loadFunc = loadFuncs[id];
-      shouldBeLoaded = loadFunc and loadFunc("ScanForLoads_Auras", incombat, vehicle, vehicleUi, inpetbattle, group, player, realm, class, spec, race, faction, playerLevel, zone, zoneId, zonegroupId, encounter_id, size, difficulty, role);
-      couldBeLoaded = loadFunc and loadFunc("ScanForLoads_Auras", true, vehicle, vehicleUi, true, group, player, realm, class, spec, race, faction, playerLevel, zone, zoneId, zonegroupId, encounter_id, size, difficulty, role);
+      local loadOpt = loadFuncsForOptions[id];
+      shouldBeLoaded = loadFunc and loadFunc("ScanForLoads_Auras", incombat, inencounter, inpetbattle, vehicle, vehicleUi, group, player, realm, class, spec, race, faction, playerLevel, zone, zoneId, zonegroupId, encounter_id, size, difficulty, role);
+      couldBeLoaded =  loadOpt and loadOpt("ScanForLoads_Auras",   nil,      nil,         nil,         nil,     nil,       nil,   player, realm, class, spec, race, faction, playerLevel, nil,  nil,    nil,         nil,          nil,  nil,        role);
```
Comment (emptyrivers) [inline WeakAuras/WeakAuras.lua]: I’m not sure if anything can be reasonably done to improve this, but I don’t really like the idea of passing a dozen nils to a function.

Comment (InfusOnWoW) [reply WeakAuras/WeakAuras.lua]: I did that on purpose, to indicate which values are ignored by loadOpt

File: WeakAurasOptions/OptionsFrames/IconPicker.lua (hunk 3 above)
Comment (InfusOnWoW) [reply WeakAurasOptions/OptionsFrames/IconPicker.lua]: Ah no it didn't change. I did change that after testing, but apparently didn't commit it.

File: WeakAuras/WeakAuras.lua (hunk 4 above)
Comment (emptyrivers) [reply WeakAuras/WeakAuras.lua]: I get that, I just wonder if there isn’t a more elegant solution. I suppose there isn’t without redesigning how ScanForLoads_Auras works. It’s fine as it is.

**Pull request merged**